prov/rxm/src/rxm_cq.c

```diff
@@ -90,9 +90,15 @@ static int rxm_match_iov(const struct iovec *iov, void **desc,
 	return FI_SUCCESS;
 }
 
+static uint64_t rxm_cq_get_rx_flags(struct rxm_rx_buf *rx_buf)
+{
+	return ((rx_buf->pkt.hdr.flags & FI_REMOTE_CQ_DATA) |
+		ofi_rx_flags[rx_buf->pkt.hdr.op]);
+}
```

Comment:
I agree, need to revise it -- we should specify it on a sender side as it was before.
Good catch! Thank you!

Reply:
I changed this function:
`rx_buf->pkt.hdr.flags & FI_REMOTE_CQ_DATA` -> `rx_buf->pkt.hdr.flags`
and I added asserts in sender side to guarantee that only `FI_REMOTE_CQ_DATA` is passed to the flags.


This function is used only for buffered receive now.


For other receive operations:
It just uses `rx_buf->pkt.hdr.flags` as it was before my changes, because `ecv_entry->comp_flags` is `FI_RECV | FI_MSG` or `FI_RECV | FI_TAGGED`. This should be better than we had earlier, because RxM set flags (`FI_MSG` or `FI_TAGGED`) on sender side and use them on receiver side

Do you agree with this approach?

Reply:
Did you already change or going to change because I don't see any changes in the function? If receiver can correctly determine the flags (FI_MSG / FI_TAGGED), the sender need not add them.